src/vs/base/common/fuzzyScorer.ts

```diff
@@ -540,11 +541,51 @@ function doScoreItemFuzzySingle(label: string, description: string | undefined,
 				}
 			});
 
-			return { score: labelDescriptionScore, labelMatch, descriptionMatch };
+			itemScore = { score: labelDescriptionScore, labelMatch, descriptionMatch };
 		}
 	}
 
-	return NO_ITEM_SCORE;
+	// Finally try to compute a score based on the path if we have one.
+	// This allows us to partially match on the absolute path of the item
+	// instead of only matching on the workspace folder relative path.
+	//
+	// We assume the label is the path's basename and the description, if
+	// present, is a subpath of the path's dirname or '.'.
+	if (path?.endsWith(label) &&
```

Comment:
I'm a bit confused on what this adds to the fuzzy scoring. It already considers the description above, which is just the path without the base name.